packages/plugins/typescript/typescript/tests/typescript.spec.ts

```diff
@@ -2678,4 +2678,71 @@ describe('TypeScript', () => {
       }
     `);
   });
+
+  it('should use implementing types as node type - issue #5126', async () => {
+    const testSchema = buildSchema(/* GraphQL */ `
+      type Matrix {
+        pills: [Pill!]!
+      }
+
+      interface Pill {
+        id: ID!
+      }
+
+      type RedPill implements Pill {
+        red: String!
+      }
+
+      type GreenPill implements Pill {
+        green: String!
+      }
+
+      interface Foo {
+        id: ID!
+      }
+
+      type Bar implements Foo {
+        lol: String!
+      }
+
+      type Hello {
+        foo: Foo!
+      }
+
+      type NoInterface {
+        hello: Hello!
+      }
+    `);
+
+    const output = (await plugin(
+      testSchema,
+      [],
+      {
+        useImplementingInterfaces: true,
+      } as any,
+      { outputFile: 'graphql.ts' }
+    )) as Types.ComplexPluginOutput;
+
+    // Type should be Array<RedPill|GreenPill> and not Pill
+    expect(output.content).toBeSimilarStringTo(`
```

Comment:
Can you please add here:

```ts
 expect(output.content).toMatchSnapshot();
```

Reply:
I just added this here but honestly I don't know what it does so you might want to take another look.